xrootd/cmd/xrd-srv/handler.go

```diff
@@ -48,3 +60,153 @@ func (h *handler) Dirlist(sessionID [16]byte, request *dirlist.Request) (xrdprot
 
 	return resp, xrdproto.Ok
 }
+
+// Open implements server.Handler.Open.
+func (h *handler) Open(sessionID [16]byte, request *open.Request) (xrdproto.Marshaler, xrdproto.ResponseStatus) {
+	var flag int
+	if request.Options&xrdfs.OpenOptionsOpenRead != 0 {
+		flag |= os.O_RDONLY
+	}
+	if request.Options&xrdfs.OpenOptionsOpenUpdate != 0 {
+		flag |= os.O_RDWR
+	}
+	if request.Options&xrdfs.OpenOptionsOpenAppend != 0 {
+		flag |= os.O_APPEND
+	}
+	if request.Options&xrdfs.OpenOptionsNew != 0 || request.Options&xrdfs.OpenOptionsDelete != 0 {
+		flag |= os.O_CREATE
+		if request.Options&xrdfs.OpenOptionsDelete == 0 {
+			flag |= os.O_EXCL
+		} else {
+			flag |= os.O_TRUNC
+		}
+	}
+
+	filePath := path.Join(h.basePath, request.Path)
+	if request.Options&xrdfs.OpenOptionsMkPath != 0 {
+		if err := os.MkdirAll(path.Dir(filePath), os.FileMode(request.Mode)); err != nil {
+			return xrdproto.ServerError{
+				Code:    xrdproto.IOError,
+				Message: fmt.Sprintf("An IO error occurred: %v", err),
+			}, xrdproto.Error
+		}
+	}
+
+	file, err := os.OpenFile(filePath, flag, os.FileMode(request.Mode))
+	if err != nil {
+		return xrdproto.ServerError{
+			Code:    xrdproto.IOError,
+			Message: fmt.Sprintf("An IO error occurred: %v", err),
+		}, xrdproto.Error
+	}
+
+	sess, _ := h.sessions.LoadOrStore(sessionID, &session{handles: make(map[xrdfs.FileHandle]*os.File)})
+	s := sess.(*session)
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	var handle xrdfs.FileHandle
+
+	for i := 0; i < 255; i++ {
+		handle[0] = byte(i)
+		for j := 0; j < 255; j++ {
+			handle[1] = byte(j)
+			if _, ok := s.handles[handle]; !ok {
```

Comment:
`s/ok/dup/g` ?
so the `!ok` feels less like an error and actually what we were looking for.